circleguard/draw.py

```diff
@@ -62,13 +62,21 @@ def run(self):
         data1 = np.transpose(data1)
         data2 = np.transpose(data2)
 
-        # create plot for each replay and add legend with player names
-        fig, ax = plt.subplots()
+        total = len(data1[0])  # stationary cause otherwise it would be a pain
 
-        plot1 = plt.plot('x', 'y', "red", animated=True, label=self.replay1.player_name)[0]
-        plot2 = plt.plot('', '', "blue", animated=True, label=self.replay2.player_name)[0]
+        def setup():
+            global plot1, plot2, fig, ax, animation
```

Comment:
using global here doesn't sit right with me, leaving this to sam's discretion because he actually knows matplotlib

Reply:
yeah try removing this because we already return animation and store it at the caller, which should be enough

Reply:
global is necessary so ax & co can be accessed outside the function, without global you get an `NameError: name 'ax' is not defined` Error